### DISCUSSION TITLE
Batch updates

### DISCUSSION THREAD

**Discussion by ZakAl-Hassani** [thepirat000/Audit.NET#644] [General]
EF Core 7.0 introduced ExecuteUpdate and ExecuteUpdateAsync methods. These do not currently seem to get picked up by audit.Net. Are there any plans to implement this or have I just missed the implementation (please point me in the right direction).

**Comment by thepirat000** [comment]
Since ExecuteUpdate / ExecuteDelete bypasses the EF change tracking, the only way to intercept those calls is by using the [command interceptor](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/README.md#low-level-command-interception), for example:

```c#
builder.Services.AddDbContext<YourContext>(_ => _
    .AddInterceptors(new AuditCommandInterceptor() { ExcludeReaderEvents = true, ... })
```

See https://github.com/thepirat000/Audit.NET/issues/607#issuecomment-1592067863
